Sources/MapboxDirections/TollPrice.swift

```diff
@@ -0,0 +1,136 @@
+import Foundation
+import Turf
+
+/**
+ :nodoc:
+ Information about toll payment method.
+ */
+public struct PaymentMethod: Hashable, Equatable {
+    /**
+     Method identifier.
+     */
+    public let identifier: String
+    
+    /**
+     Payment is done by electronic toll collection.
+     */
+    public static let electronicTollCollection = PaymentMethod(identifier: "etc")
+    /**
+     Payment is done by cash.
+     */
+    public static let cash = PaymentMethod(identifier: "cash")
+}
+
+/**
+ :nodoc:
+ Categories by which toll fees are divided.
+ */
+public struct Category: Hashable, Equatable {
+    /**
+     Category name.
+     */
+    public let name: String
+    
+    /**
+     A small sized vehicle.
+     
+     Vehicle size is [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public static let small = Category(name: "small")
+    /**
+     A standard sized vehicle.
+     
+     Vehicle size is [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public static let standard = Category(name: "standard")
+    /**
+     A middle sized vehicle.
+     
+     Vehicle size is [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public static let middle = Category(name: "middle")
+    /**
+     A large sized vehicle.
+     
+     Vehicle size is [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public static let large = Category(name: "large")
+    /**
+     A jumbo sized vehicle.
+     
+     Vehicle size is [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public static let jumbo = Category(name: "jumbo")
+}
+
+public typealias CategoriesTolls = [Category: Decimal]
+public typealias PaymentMethods = [PaymentMethod: CategoriesTolls]
```

Comment:
Do we need these type aliases? The interface might be clearer if we simply inline these types.

Reply:
I am not sure that this is easier to read:
```
public let paymentMethods: [[TollPaymentMethod: [TollCategory: Decimal]]]
```

Reply:
I'm not fond of embedding that much detail into a type or variable name, IMHO that looks synthetic. Probably it would be better to generalize the contents and name this typealias something like `PaymentDetails`? Thus it suggests that there are various parameters inside, which may or may not be grouped by category, payment method, etc.

Reply:
It is an awkward name, but probably also a sign that the raw structure exposed by the JSON response is not an ideal structure to expose to developers in Swift.

I hesitate to rabbit-hole on what’s still an undocumented part of the API, but maybe an alternative would be a flatter array of TollPrice structures, where TollPrice itself can contain a vehicle type condition and a payment method condition, both as properties. When encoding, we can assemble the elements of this array into a dictionary, throwing if there are duplicates. It reminds me of how a `URLComponents` has an array of `URLQueryItem`s even though typically one thinks of it as a key-value store.

I realize this goes against some of what I wrote in https://github.com/mapbox/mapbox-directions-swift/pull/749#discussion_r982701348, when I thought we could get away with a rawer structure. The downside is that we would shift some consistency checks to runtime, but the upside is that if the API introduces more axes in the future, we wouldn’t have to dramatically restructure the property and all its call sites.

If this doesn’t sound like a sound strategy, then at least we should rename the property to describe the whole type, not just the keys: `pricesByPaymentMethod`, `paymentsByMethod`, or… `userInfo`. 😉

Reply:
Could you please expand what you mean by the need to restructure properties if there are new entities added? As I see, current structure is extendable with new types of payment methods and categories as well as adding new properties on those levels.
Flattening the structure, on the other hand, creates troubles coding foreign keys which almost the same as addition of new properties to JSON. I agree that flat structure might be more readable for user, and `paymentMethods` should be renamed for the same reason, but I think that altering the hierarchy here may damage extensibility.
BTW I still think `PaymentMethods -> PaymentDetails` is a viable option :D